proxy/http/HttpSM.cc

```diff
@@ -3965,13 +3966,58 @@ HttpSM::state_remap_request(int event, void * /* data ATS_UNUSED */)
   return 0;
 }
 
+// This check must be called before remap.  Otherwise, the client_request host
+// name may be changed.
+void
+HttpSM::check_sni_host()
+{
+  // Check that the SNI and host name fields match, if it matters
+  // Issue warning or mark the transaction to be terminated as necessary
+  int host_len;
+  const char *host_name = t_state.hdr_info.client_request.host_get(&host_len);
+  if (host_name && host_len) {
```

Comment:
I know it's late, but why would we nest this many levels, rather than merging it all into one if()? There are no else's here ...

Reply:
I guess it's because there are things to do between the if-s and gathering all information at the beginning is costly or impossible. I'd separate the if-s and put return on each checks.

For example:
```
A = X.getA();
if (!A) {
  return;
}

B = Y.getB(A)
if (!B) {
  return;
}
...
```
Technically it can be combined like `(A = X.getA()) && (B = Y.getB(A))` but it'd be more difficult to read.